CHANGELOG.md

```diff
@@ -35,7 +35,25 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 # Changelog
 
-## [Unreleased]
+## [v0.38.5] - TBD
+
+### Improvements
+
+* (deps) Bump Tendermint version to [v0.33.5](https://github.com/tendermint/tendermint/releases/tag/v0.33.5)
+* (deps) Bump IAVL version to [v0.14.0](https://github.com/cosmos/iavl/releases/tag/v0.14.0)
+
+### Bug Fixes
+
+* (store) Revert IAVL pruning functionality introduced in [v0.13.0](https://github.com/cosmos/iavl/releases/tag/v0.13.0),
```

Comment:
@erikgrinaker can you recommend or add a quick note here on the topic of potential data loss or corruption please?

Reply:
Yes, I'll write specific instructions tomorrow after verifying a few things.

Reply:
Ok, here's a draft - we should test these, and consider including a repair utility. Note that if using the default settings, upgrading has a significant chance of causing data corruption (since the old default is KeepEvery:100 and the new is KeepEvery:500).

---

When upgrading from 0.38.x it is important to follow the instructions below, to prevent data loss and database corruption. Upgrading blindly with the default pruning settings has a significant chance of causing data corruption.

Do not modify pruning settings with <=0.38.4 as that may cause data corruption - the following assumes pruning settings have not been modified since the node started using 0.38.x. The default pruning setting `syncable` used `KeepEvery:100`.

* If using `KeepEvery:1` (pruning setting `nothing`), upgrading to 0.38.5 is safe.

* Otherwise, if possible, halt block processing immediately after committing a height divisible by `KeepEvery` - e.g. at block 147600 with `KeepEvery:100`. The node must _never_  have processed a height beyond that at any time in its past. Upgrading to 0.38.5 is then safe.

  [this assumes all IAVL stores are at the same version: is this always true, even when past commits have been partial?]

* Otherwise, set the 0.38.5 `KeepEvery` setting to the same as the previous `KeepEvery` setting (<=0.38.4 defaults to `KeepEvery:100` while 0.38.5 defaults to `KeepEvery:500`). Upgrading to 0.38.5 is then safe as long as you allow one `KeepEvery` interval to pass plus one `KeepRecent` interval plus one `Interval` pruning interval before changing pruning settings or deleting the last <=0.38.4 height. No versions should exist between the last height persisted with <=0.38.4 and next `KeepEvery` interval, e.g. between 147600 and 147700.

  [This can be avoided by changing the new defaults to `KeepEvery: 100`. The failure mode is that when deleting e.g. height 147600 at height 147700 it may corrupt heights between 147600-146999 causing panics and node crashes, and these heights are available by default due to `KeepRecent: 100`]

* Otherwise, make sure the last version persisted with <=0.38.4 is never deleted after upgrading to 0.38.5, as doing so may cause data loss and data corruption. This will happen if using the default pruning settings, so do not use the defaults.

* Otherwise, consider syncing the node from scratch with 0.38.5.

Reply:
@erikgrinaker feel free to directly commit to the changelog.

Reply:
Sure. Is changing the default `KeepEvery` to 100 an option? That would allow people using default settings to blindly upgrade without things breaking. Also, we don't want to include a repair utility?

Reply:
Sure, just create a PR on `master` to change the `default` option and I'll cherry-pick it once merged. I'm sure we'll want to include a repair utility. Not sure how that'll be communicated. We can add a note to the changelog to a WIKI page or something.

Reply:
#6520, will update changelog once it's cherry-picked.

I'll start working on a utility, but may need a hand to correctly detect all IAVL stores for all modules.